Cython/CodeWriter.py

```diff
@@ -519,3 +519,276 @@ def visit_CFuncDefNode(self, node):
 
     def visit_StatNode(self, node):
         pass
+
+
+class ExpressionWriter(TreeVisitor):
+
+    def __init__(self, result=None):
+        super(ExpressionWriter, self).__init__()
+        if result is None:
+            result = u""
+        self.result = result
+
+    def write(self, tree):
+        self.visit(tree)
+        return self.result
+
+    def put(self, s):
+        self.result += s
```

Comment:
Just commenting. I guess it's ok for small stuff, but I'd always go with a list in such a case since you never know what people end up using it for.